x-pack/plugins/observability_solution/infra/common/locators/asset_details_flyout_locator.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { SerializableRecord } from '@kbn/utility-types';
+import rison from '@kbn/rison';
+import { LocatorDefinition, LocatorPublic } from '@kbn/share-plugin/common';
+import { InfraLocatorDependencies } from '.';
+
+export type AssetDetailsFlyoutLocator = LocatorPublic<AssetDetailsFlyoutLocatorParams>;
+
+export type AssetDetailsFlyoutLocatorDependencies = InfraLocatorDependencies;
+
+export interface AssetDetailsFlyoutLocatorParams extends SerializableRecord {
+  tableProperties: {
+    detailsItemId: string;
+    pagination: {
+      pageIndex: number;
+      pageSize: number;
+    };
+    sorting: {
+      direction?: string;
+      field: string;
+    };
+  };
+  assetDetails: {
+    tabId?: string;
+    dashboardId?: string;
+    dateRange?: {
+      from: string;
+      to: string;
+    };
+  };
+}
+
+export const ASSET_DETAILS_FLYOUT_LOCATOR_ID = 'ASSET_DETAILS_FLYOUT_LOCATOR';
+
+export class AssetDetailsFlyoutLocatorDefinition
```

Comment:
I think the best place to put these locators would be `observability_shared` plugin, don't you think? If another plugin needs to link to these URLs they are already on a shared place.

Reply:
I added in the past a few Profiling locators here https://github.com/cauemarcondes/kibana/tree/2ffc34893965aca080078b2d10b75ec73c14afe3/x-pack/plugins/observability_solution/observability_shared/public/locators, maybe you can just follow the same thing

Reply:
Done 👍